bids/variables/variables.py

```diff
@@ -448,12 +448,13 @@ def resample(self, sampling_rate, inplace=False, kind='linear'):
         self.index = self._build_entity_index(self.run_info, sampling_rate)
 
         x = np.arange(n)
-        num = int(np.ceil(n * sampling_rate / old_sr))
+        num = len(self.index)
```

Comment:
I worry that this will hide other potential edge cases where there's a bigger mismatch... I suggest checking for a discrepancy and issuing a warning either if any is found, or if a large one is found (like you did in one of the other PRs).

Reply:
I think that was an earlier version of this PR. I'm curious what kind of edge case you're talking about?

As I understand it, we have a run duration, which is the original TR * nvols.

For a given sampling rate, we find the nearest n such that `n / sampling_rate == duration`. So this line was a way of taking from `duration ~= n_old / sr_old ~= n_new / sr_new`, and calculating `n_new` from `n_old`, `sr_old`, and `sr_new`. Index instead calculates directly from `duration` and `sr_new`. So the discrepancy we were finding is when you get an off-by-one between the two ways of calculating `n_new`.

So it's not really clear where another discrepancy can arise.